src/macaron/malware_analyzer/pypi_heuristics/sourcecode/suspicious_setup.py

```diff
@@ -123,3 +124,88 @@ def analyze(self, api_client: PyPIRegistry) -> tuple[HeuristicResult, dict]:
         if suspicious_setup:
             return HeuristicResult.FAIL, {"import_module": imports}
         return HeuristicResult.PASS, {"import_module": imports}
+
+    def extract_from_ast(self, source_content: str) -> set[str]:
+        """Extract imports from source code using the parsed AST.
+
+        Parameters
+        ----------
+        source_content: str
+            The source code as a string.
+
+        Returns
+        -------
+        set[str]
+            The set of imports.
+
+        Raises
+        ------
+        SyntaxError
+            If the code could not be parsed.
+        """
+        imports = set()
+        tree = ast.parse(source_content)
+        for node in ast.walk(tree):
+            if isinstance(node, ast.Import):
+                for alias in node.names:
+                    imports.add(alias.name)
+            elif isinstance(node, ast.ImportFrom):
+                module = node.module
+                if node.level != 0 or not module:
+                    # Relative imports are ignored.
+                    continue
```

Comment:
Why did we decide to ignore relative imports (where the original implementation accepted them)?